src/LoginForm.test.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+import { render, fireEvent } from '@testing-library/react';
+
+import LoginForm from './LoginForm';
+
+describe('LoginForm', () => {
+  const handleSubmit = jest.fn();
+  const handleChange = jest.fn();
+
+  beforeEach(() => {
+    handleSubmit.mockClear();
+    handleChange.mockClear();
+  });
+
+  function renderLoginForm({ email = '', password = '' }) {
+    return render(
+      <LoginForm
+        email={email}
+        password={password}
+        onSubmit={handleSubmit}
+        onChange={handleChange}
+      />,
+    );
+  }
+
+  it('renders input controls', () => {
+    const controls = [
+      { label: 'E-mail', value: 'email@test.com' },
+      { label: 'Password', value: 'password' },
+    ];
+
+    const { getByLabelText } = renderLoginForm(
+      {
+        email: controls[0].value,
+        password: controls[1].value,
+      },
+    );
```

Comment:
```suggestion    const email = 'test@test';
    const password = '1234';

    const { getByLabelText } = renderLoginForm({ email, password });

    const controls = [
      { label: 'E-mail', value: email },
      { label: 'Password', value: password },
    ];

```

이렇게 하면 번거롭게 배열 인덱스로 접근하지 않아도 될 겁니다.

그리고 읽는 입장에서 초기 조건이랑 변화가 자연스럽게 읽힐 것 같네요.

Reply:
그렇네요, 기존에는 어떤 값들로 rendering 되는지 한번 더 생각해야 하는 느낌인데,
제안 주신 것은 직관적으로 알 수 있는 것 같습니다.